src/core/mac/mac.cpp

```diff
@@ -569,6 +596,12 @@ void Mac::ScheduleNextTransmission(void)
     else
     {
         mState = kStateIdle;
+
+        if (mDisableRadioOnIdle)
+        {
+            mDisableRadioOnIdle = false;
+            DisableRadio();
+        }
```

Comment:
What if the caller calls `otInterfaceUp` and then `otInterfaceDown` while the scan is active? It would be easy enough to clear the `mDisableRadioOnIdle` flag if `otInterfaceUp` is called, but more complicated to try to track a number of 'references' to the radio to determine if it should be up or not.

Is this practically a scenario we need/want to support?